### PR TITLE
Multiuser fixes

### PR DESCRIPTION
**Pull request by rdunklau**
Small fixes regarding multiple user and performance problems with quals.

### CODE REVIEW

**Comment by rjuju** [comment]
@rdunklau last commit should fix the already existing issue I was talking about.

File: powa/qual.py [hunk 1]
```diff
@@ -72,23 +78,28 @@ def get(self, database, query, qual):
...
+                    "queryids": [query],
                     "qualid": qual}))
         my_qual = None
         other_queries = {}
+        other_queries = {r['queryid']: r['query'] for r in self.execute(text("""
+            SELECT DISTINCT queryid, query
+            FROM powa_qualstats_quals
+            JOIN powa_statements USING (queryid)
+            WHERE qualid = :qual
+            LIMIT 5"""), {"qual": qual})}
```
Comment (rjuju) [inline powa/qual.py]: this **5** seems a little bit arbitrary.  As far as I can tell it shouldn't raise performance issues to try to retrieve more queries involving the same qual, and it seems like an interesting information to have, especially if there are more than 5 queries having the same qual.  What about raising it quite a bit (50 or so?)

Also, shouldn't the query exclude the current queryid?

File: powa/qual.py [hunk 2]
```diff
@@ -72,23 +78,28 @@ def get(self, database, query, qual):
...
+            .column((stmt.c.queryid == bindparam("query")).label("is_my_query")))
         quals = list(self.execute(
             stmt,
             params={"query": query,
                     "from": self.get_argument("from"),
                     "to": self.get_argument("to"),
+                    "queryids": [query],
                     "qualid": qual}))
         my_qual = None
         other_queries = {}
```
Comment (rjuju) [inline powa/qual.py]: is it still needed?

File: powa/qual.py (hunk 1 above)
Comment (rdunklau) [reply powa/qual.py]: Performance wise, it's not a problem but the current UI makes you scroll for kilometers before seeing any other information.
Those queries should probably moved to a separate grid widget, but in the meantime that seemed like the best option.

Comment (rjuju) [reply powa/qual.py]: What about adding an overflow attribute to the div?  It'll also fix the case when you have only few queries, but super long queries.  Or truncate the query text as in the per database query list.

And what about adding a "queryid != :query"?

File: powa/query.py [hunk 3]
```diff
@@ -122,25 +122,25 @@ def bps(col):
...
             # aligned to kernel ticks. As such, we have to clamp values to 100%
             total_time_percent = lambda x: least(100, (x * 100) /
                                                  total_time)
             cols.extend([
-                kc.reads,
-                kc.writes,
-                total_time_percent(kc.user_time * 1000).label("user_time"),
-                total_time_percent(kc.system_time * 1000).label("system_time"),
+                sum(kc.reads).label("reads"),
+                sum(kc.writes).label("write"),
```
Comment (anayrat) [inline powa/query.py]: shouldn't be `label("writeS")` ?

Comment (rjuju) [reply powa/query.py]: oops, indeed :)  good catch!  fixed and forced-pushed

Comment (anayrat) [reply powa/query.py]: Thanks

File: powa/qual.py (hunk 1 above)
Comment (rdunklau) [reply powa/qual.py]: I'll do that

File: powa/qual.py (hunk 2 above)
Comment (rdunklau) [reply powa/qual.py]: I'll check that

**Comment by rjuju** [comment]
@rdunklau any news?  I'd like to merge this as soon as possible.

File: powa/query.py [hunk 4]
```diff
@@ -209,28 +209,32 @@ def per_sec(col):
...
                      else_=disk_hit_ratio).label("disk_hit_ratio"),
                 case([(total_blocks == 0, 0)],
                      else_=sys_hitratio).label("sys_hit_ratio")])
             from_clause = from_clause.join(
                 kcache_query,
-                kcache_query.c.ts == c.ts)
+                and_(kcache_query.c.ts == c.ts,
+                     kcache_query.c.queryid == c.queryid,
+                     kcache_query.c.userid == c.userid,
+                     kcache_query.c.dbid == c.dbid))
```
Comment (rjuju) [inline powa/query.py]: dbid isn't part of kcache_query.

**Comment by rjuju** [comment]
@rdunklau I pushed two commits to fix the issues I saw for both per-query page and wizard.  Can you check if that's ok and do additional testing?

**Comment by rjuju** [comment]
Rebased against latest master and merged.

**Pull request merged**